packages/react-performance/README.md

```diff
@@ -0,0 +1,261 @@
+
+```tsx
+// LastNavigationDetails.tsx
+import React, {useState} from 'react';
+import {useNavigationListener, Navigation} from '@shopify/react-performance';
+
+export function LastNavigationDetails() {
+  const [lastNavigation, setLastNavigation] = useState<Navigation | null>(null);
+
+  useNavigationListener(navigation => {
+    setLastNavigation(navigation);
+  });
+
+  if (lastNavigation == null) {
+    return <p>no data</p>;
+  }
+
+  const {duration, isFullPageNavigation, target};
+  const navigationType = isFullPageNavigation
+    ? 'full page navigation'
+    : 'single-page-app style navigation';
+  return (
+    <p>
+      The last navigation was to {target}. It was a {navigationType} navigation
+      which took {duration / 1000} seconds to complete.
+    </p>
+  );
+}
+```
+
+We can render this component anywhere in our tree, but lets do so in our App component from earlier.
+
+```tsx
+// App.tsx
+import React from 'react';
+import {Performance, PerformanceContext} from '@shopify/react-performance';
+import {LastNavigationDetails} from './LastNavigationDetails';
+
+// in a Server-Side-Rendering enabled app you will likely only want to instantiate this if `document` is defined.
+const performance = new Performance();
+
+function App() {
+  return (
+    <PerformanceContext.Provider value={performance}>
+      {/* The rest of your app */}
+      <LastNavigationDetails />
+    </PerformanceContext.Provider>
+  );
+}
+```
+
+Now our developers will have access to up-to-date information about how long initial page-loads and incremental navigations take to complete without needing to open their devtools or dig into DOM APIs manually.
+
+### Reporting data to the server
+
+Performance metrics data is most useful when it can be aggregated and used to make smart optimization decisions or warn you when your app is beginning to become slow. To aggregate such data you'll need to be able to send it to a server. `@shopify/react-performance` provides the `PerformanceReport` component to facilitate this.
+
+> This section is a work in progress. It will be filled out as helper libraries for marshalling data on the server are added to Quilt.
+
+## API
+
+Most APIs provided by this package are available as both hooks (ie. `useNavigationListener`), as well as components (ie. `<NavigationListener />`).
+
+This package also re-exports all of the API of [`@shopify/performance`](https://www.npmjs.com/package/@shopify/performance).
+
+### Hooks
+
+#### useLifecycleEventListener
+
+A custom hook which takes in callback to invoke whenever the `Performance` context object emits a lifecycleEvent.
+
+```tsx
+import {useLifecycleEventListener} from '@shopify/react-performance';
+
+function SomeComponent() {
```

Comment:
I would almost just skip this example all together. It does not feel very useful till it is compose together later with more meaningful example that show case where this should go in an app and the reason why.

Reply:
hmm, I'm not sure what the best choice is here. I like having something you can see that shows the API, and I don't think going into how everything fits together makes sense in the API section. I do want to add more to the first section (quick start) that talks about how to hook things up to server-side reporting in the future though. :/

Reply:
> I like having something you can see that shows the API

me too! 
however in this case some foresight of where it should be place is not as clear, and the example code so far feel like if you know react you can get that far.

if we would to keep the example, I would like to see a more involved `onEvent` in life cases.